domains/grid/utils/gridParser.ts

```diff
@@ -108,21 +108,44 @@ const parseSpotifyWidgetInput = (input: string): LayoutItemExtended | never => {
   throw new Error('Invalid Spotify input')
 }
 
-const parseSoundCloudWidgetInput = (
+const parseSoundCloudWidgetInput = async (
+  input: string
+): Promise<LayoutItemExtended> => {
+  const SOUNDCLOUD_SHARE_URL_REGEX =
+    /https:\/\/soundcloud\.com\/([a-zA-Z0-9_-]+)(?:\/(sets\/[a-zA-Z0-9_-]+|[a-zA-Z0-9_-]+))?\/?/
+
+  // Check if the input already contains an embed URL
+  try {
+    return parseSoundCloudWidgetInputFromEmbed(input)
+  } catch {}
+
+  // Check if the input is a SoundCloud share URL
+  const [shareUrlMatch] = input.match(SOUNDCLOUD_SHARE_URL_REGEX) || []
+  if (shareUrlMatch) {
+    const soundCloudEmbed = await getSoundCloudEmbedUrl(shareUrlMatch)
+
+    if (soundCloudEmbed) {
+      return parseSoundCloudWidgetInputFromEmbed(soundCloudEmbed)
+    }
+  }
+
+  throw new Error('Invalid SoundCloud input')
+}
+
+export const parseSoundCloudWidgetInputFromEmbed = (
```

Comment:
I guess we don't need to export this as it's just for internal use.

Reply:
its exported because its covered by some tests